src/main/java/com/metamx/metrics/cgroups/SigarPidDiscoverer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.metamx.metrics.cgroups;
+
+import com.metamx.metrics.SigarUtil;
+import org.hyperic.sigar.Sigar;
+
+public class SigarPidDiscoverer implements PidDiscoverer
+{
+  private final Sigar sigar = SigarUtil.getSigar();
+
+  @Override
+  public long getPid()
+  {
+    return sigar.getPid();
```

Comment:
Yes, I was torn here for this one. On one hand this uses a single Sigar object, on the other hand this keeps an extra Sigar object around instead of creating a new one every call. I'll use the util method and allow the util method to do any object optimizations as needed.